src/user_summary/static/user_summary/js/main.js

```diff
@@ -0,0 +1,909 @@
+    }
+}
+
+/* Function to return the correct image for badge */
+function badgeMedalImageSource(type) {
+    switch (type) {
+    case 'gold':
+        return '/outreachy-wikicv/static/user_summary/img/goldmedal1.png';
+    case 'silver':
+        return '/outreachy-wikicv/static/user_summary/img/silvermedal1.png';
+    case 'bronze':
+        return '/outreachy-wikicv/static/user_summary/img/bronzemedal1.png';
+    }
+}
+
+$(document).ready(function () {
+
+    // Acquiring csrf token
+    var csrftoken = getCookie('csrftoken');
+
+    // Making sure csrf token is sent with every ajax request
+    $.ajaxSetup({
+        beforeSend: function (xhr, settings) {
+            if (!csrfSafeMethod(settings.type) && !this.crossDomain) {
+                xhr.setRequestHeader("X-CSRFToken", csrftoken);
+            }
+        }
+    });
+
+    google.charts.load('current', {'packages': ['corechart', 'calendar',
+            'bar']});
+
+    // Ajax call to update data for username on server
+    $.ajax({url: "/outreachy-wikicv/update-cached-data/" + String(username),
+            success: function (data) {
+        if (data.result) {
+            alert("Your data has been updated on server." +
+                    " Please refresh the page.");
+        }
+    }});
+
+/*
+ Whenever time filter is changed for activity and spread over projects,
+ change the graphs
+*/
+    $('input.time-filter').on('change', function () {
+        $('input.time-filter').not(this).prop('checked', false);
+        $.ajax({url: "/outreachy-wikicv/load-graphs/",
+                type: "get",
+                data: {
+            username: String(username),
+            filter: String(this.value),
+            time_filter_type: "general"
+        },
+                success: function (data) {
+            google.charts.setOnLoadCallback(function () {
+                drawPieChart(data.spread_over_projects_data);
+                drawCalendarChart(data.created_activity_chart_data,
+                        'Articles Created', 'articles-created');
+                drawCalendarChart(data.edits_activity_chart_data,
+                        'Articles Edited', 'articles-edited');
+            });
+        },
+                error: function () {
+            $("#alert-message-holder").html(
+                "<div class='alert alert-danger' role='alert'><strong>" +
+                "Oh snap!</strong>Something went wrong while creating" +
+                " graphs! Please report Megha at " +
+                "meghasharma4910@gmail.com</div>"
+            );
+        }});
+    });
+
+
+    //Whenever time filter is changed for impact, change the graph
+    $('input.impact-time-filter').on('change', function () {
+        $('input.impact-time-filter').not(this).prop('checked', false);
+        $.ajax({url: "/outreachy-wikicv/load-graphs/",
+                type: "get",
+                data: {
+            username: String(username),
```

Comment:
I don't think explicit casts have any effect, HTTP request data is always a string.

Reply:
Okay.